### ISSUE TITLE
Testar modelos few-shot

### ISSUE THREAD

**Issue by Leanddro13** [ResidenciaTICBrisa/07_ProcessamentoLinguagemNatural#33]
# Descrição
Após estudar os modelos de few-shot, implementar eles.

# Tarefas
- [x] Testar capacidade de generalização dos modelos
- [ ] Verificar acurácia dos modelos

# Critérios de Aceitação
- [ ] Branch com as alterações pedidas aprovada pelo grupo

**Comment by LexTOliver** [comment]
Alguns estudos separados para leitura e compreensão do tema "Few-Shot Text Classification" e formas de implementação:

- [Diverse Few-Shot Text Classification with Multiple Metrics](https://paperswithcode.com/paper/diverse-few-shot-text-classification-with) (2018)
- [Few-Shot Text Classification with Pre-Trained Word Embeddings and a Human in the Loop](https://paperswithcode.com/paper/few-shot-text-classification-with-pre-trained) (2018)
- [Induction Networks for Few-Shot Text Classification](https://paperswithcode.com/paper/few-shot-text-classification-with-induction) (2019)
- [Few-shot Text Classification with Distributional Signatures](https://paperswithcode.com/paper/few-shot-text-classification-with) (2020)
- [Exploiting Cloze Questions for Few Shot Text Classification and Natural Language Inference](https://paperswithcode.com/paper/exploiting-cloze-questions-for-few-shot-text) (2020)
- [Using Triplet Loss and Siamese Neural Networks to Train Catalog Item Embeddings](https://doordash.engineering/2021/09/08/using-twin-neural-networks-to-train-catalog-item-embeddings/) (2021)
- [Few-Shot Learning with Siamese Networks and Label Tuning](https://paperswithcode.com/paper/few-shot-learning-with-siamese-networks-and-1) (2022)

**Comment by LexTOliver** [comment]
Alguns algoritmos para testar:

Dos artigos acima:
- [Induction Networks](https://github.com/zhongyuchen/few-shot-text-classification)
- [Distributional Signatures](https://github.com/YujiaBao/Distributional-Signatures)
- [Pattern-Exploiting Training (PET)](https://github.com/timoschick/pet)

Dos artigos da issue passada:
- [Few Shot Text Classification - Cloudera](https://github.com/fastforwardlabs/few-shot-text-classification)
- [Text classification from few training examples](https://maelfabien.github.io/machinelearning/NLP_5/#implementation)

**Comment by LexTOliver** [comment]
A maioria dos modelos encontrados ainda estavam em estágio de pesquisa e exploração da task few-shot. Nominalmente, existem poucos modelos marcantes conhecidos que podem ser reproduzidos. Dentre eles: redes prototípicas, redes siamesas, SetFit. Todos baseando-se nos conceitos de codificação do texto em embeddings e classificação através da similaridade por cosseno.

**Comment by LexTOliver** [comment]
Dentre os modelos mais conhecidos, ainda existem pouca documentação e implementação disponível para reproducibilidade, o que necessitou uma estudo mais aprofundado dos seus conceitos e técnicas e atrasou um pouco o cronograma de entrega das issues.

**Comment by LexTOliver** [comment]
O melhor modelo que foi selecionado para foco e teste foi o [SetFit](https://huggingface.co/blog/setfit), disponível em versão early no [Pypi](https://pypi.org/project/setfit/) com uma [documentação](https://github.com/huggingface/setfit/tree/main/notebooks) satisfatória e vídeos tutorias explicando sua implementação e uso.

**Comment by LexTOliver** [comment]
Os testes do SetFit se encontram no [Colab](https://colab.research.google.com/drive/1FE8fQ3oVX1eOw-KP76UvKaI1Waje9pS0?usp=sharing).

**Comment by LexTOliver** [comment]
Os códigos de teste estão disponíveis no Colab.
